src/Domain/Details.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain;
+
+final class Details
+{
+    /** @var mixed */
+    private $original;
+
+    /** @var string */
+    private $file;
+
+    /** @var int */
+    private $line;
+
+    /** @var string  */
+    private $function;
+
+    /** @var string */
+    private $message;
+
+    public static function make(): Details
+    {
+        return new Details();
+    }
+
+    public function withFile(string $file): Details
+    {
+        $this->file = $file;
+        return $this;
+    }
+
+    /**
+     * @param mixed $original
```

Comment:
You have no idea about what is coming for `original`?

Reply:
Well I can add the Sniff error class and keep mixed?

As right now the only data which is added in here is actually the sniff error, but for example with the phpstan error class a new type is added to this also.

We could also remove the original all together as we actually don't use it anywhere yet?